libs/telemetry/src/registry.cpp

```diff
@@ -183,11 +124,36 @@ void Registry::Collect(std::ostream &stream)
   OutputStream telemetry_stream{stream};
 
   std::lock_guard<std::mutex> guard(lock_);
-  for (auto const &measurement : measurements_)
+  for (auto const &named_cell : measurements_)
   {
-    measurement->ToStream(telemetry_stream);
+    for (auto const &measurement : named_cell.second)
+    {
+      measurement->ToStream(telemetry_stream);
+    }
   }
 }
 
+std::size_t Registry::FastMeasurementHash::operator()(MeasurementPtr const &measurement) const
+{
+  assert(measurement);
+
+  auto const &labels = measurement->labels();
+
+  std::string flattened = std::accumulate(
+      begin(labels), end(labels), std::string(), [](std::string accum, auto &&element) {
+        return std::move(accum) + element.first + KEY + element.second + VALUE;
```

Comment:
Do we not need to sort the keys here first to ensure a consistent hash. Also it might be worth (maybe in the future) to switching to a XOR based hash aggregation as the performance of this with multiple labels might be expensive